tests/unit/test_exceptions.py

```diff
@@ -79,6 +79,12 @@ def testCodeInvariants(self):
 
     def testInstantiation(self):
         for class_ in self._getExceptionClasses():
+            # some exceptions are becoming too complicated to instantiate
+            # like the rest of the exceptions; just do them manually
+            if class_ in [
+                    exceptions.RequestValidationFailureException,
+                    exceptions.ResponseValidationFailureException]:
+                continue
```

Comment:
Agreed.  Continues are worse than gotos. At least gotos have an obvious label.

I simplify stuff like this by breaking the loop in to functions.

Jerome Kelleher <notifications@github.com> writes:
> In tests/unit/test_exceptions.py:
> 
> > @@ -79,6 +79,12 @@ def testCodeInvariants(self):
> >
> >      def testInstantiation(self):
> >          for class_ in self._getExceptionClasses():
> > +            # some exceptions are becoming too complicated to instantiate
> > +            # like the rest of the exceptions; just do them manually
> > +            if class_ in [
> > +                    exceptions.RequestValidationFailureException,
> > +                    exceptions.ResponseValidationFailureException]:
> > +                continue
> 
> continues make me ill; can't we work around this, and keep the nice loop?
> What's the issue that we hit with these two?
> 
> —
> Reply to this email directly or view it on GitHub.*
>